pkg/process/util/ipc.go

```diff
@@ -0,0 +1,49 @@
+package util
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/api/pb"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials"
+)
+
+// GetDDAgentClient creates a pb.AgentClient for IPC with the main agent via gRPC. This call is blocking, so
+// it is up to the caller to supply a context with appropriate timeout/cancel options
+func GetDDAgentClient(ctx context.Context) (pb.AgentClient, error) {
```

Comment:
This seems like it could be shared with other teams, do we really want it in `pkg/process`?

Reply:
I'd probably agree with this as well

Reply:
Any thoughts on where this should go?

Reply:
Something like `/pkg/util/grpc`, `/pkg/util/ipc`, or something similar maybe?